executor/executor.go

```diff
@@ -1931,3 +1931,41 @@ func (e *DummyScanExec) Close() error {
 func (e *DummyScanExec) Next() (*Row, error) {
 	return nil, nil
 }
+
+// TempStoreExec represents TempStore executor.
+// it stores the return values of the executor of its child node.
+type TempStoreExec struct {
+	schema     expression.Schema
+	Src        Executor
+	storedRows []*Row
+	cursor     int
+	closed     bool
+}
+
+// Schema implements the Executor Schema interface.
+func (e *TempStoreExec) Schema() expression.Schema {
+	return e.schema
+}
+
+// Close implements the Executor Close interface.
+func (e *TempStoreExec) Close() error {
+	e.cursor = 0
+	if !e.closed {
+		e.closed = true
+		return e.Src.Close()
+	}
+	return nil
+}
+
+// Next implements the Executor Next interface.
+func (e *TempStoreExec) Next() (*Row, error) {
+	if e.cursor == len(e.storedRows) {
```

Comment:
This part makes me confused. You can write like this : if not closed, prepare the storedRows and return srcRow, if the row from src is nil, set closed true and close the src. If closed is true, use the increasing cursor to read data. If cursor is out of index, return nil.